utils/kubernetes/kube_state_processor.py

```diff
@@ -120,6 +167,43 @@ def kube_node_status_allocatable_pods(self, message, **kwargs):
             tags = ['node:{}'.format(self._extract_label_value("node", metric.label))]
             self.gauge(metric_name, val, tags)
 
+    # Labels attached: daemonset, namespace
+    def kube_daemonset_status_current_number_scheduled(self, message, **kwargs):
+        """ Number of nodes running at least 1 daemon pod that are supposed to run one. """
+        metric_name = NAMESPACE + '.daemonset.current_scheduled'
+        for metric in message.metric:
+            val = metric.gauge.value
+            tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
+            self.gauge(metric_name, val, tags)
+
+    # Labels attached: daemonset, namespace
+    def kube_daemonset_status_number_misscheduled(self, message, **kwargs):
+        """ Number of nodes running at least 1 daemon pod that are NOT supposed to run one. """
+        metric_name = NAMESPACE + '.daemonset.misscheduled'
+        for metric in message.metric:
+            val = metric.gauge.value
+            tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
+            self.gauge(metric_name, val, tags)
+
+    # Labels attached: daemonset, namespace
+    def kube_daemonset_status_desired_number_scheduled(self, message, **kwargs):
+        """ Number of nodes that should be running a daemon pod. """
+        metric_name = NAMESPACE + '.daemonset.desired_scheduled'
+        for metric in message.metric:
+            val = metric.gauge.value
+            tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
+            self.gauge(metric_name, val, tags)
+
+    # Labels attached: daemonset, namespace
+    def kube_daemonset_metadata_generation(self, message, **kwargs):
```

Comment:
I don't think this one is interesting to collect, do you have a use for it in mind?